tsl/src/dist_backup.c

```diff
@@ -115,6 +118,18 @@ create_distributed_restore_point(PG_FUNCTION_ARGS)
 					 errhint("Connect to the access node and create the distributed restore point "
 							 "from there.")));
 
+		/* Get a list of data nodes and ensure all of them are available */
+		data_node_list = data_node_get_node_name_list_with_aclcheck(ACL_NO_CHECK, false);
+		foreach (lc, data_node_list)
+		{
+			const char *node_name = lfirst(lc);
+			const ForeignServer *server;
+
+			server = data_node_get_foreign_server(node_name, ACL_NO_CHECK, false, false);
+			if (!ts_data_node_is_available_by_server(server))
+				ereport(ERROR, (errmsg("some data nodes are not available")));
+		}
```

Comment:
I suggest making this a utility function. For example:

data_node_fail_if_nodes_are_unavailable();